src/JoeScan.Pinchot.Shared/DataType.cs

```diff
@@ -29,7 +29,8 @@ internal static class DataTypeValues
         internal static readonly IEnumerable<DataType> DataTypes =
             Enum.GetValues(typeof(DataType))
                 .Cast<DataType>()
-                .Where(d => d != DataType.Invalid);
+                .Where(d => d != DataType.Invalid)
+                .ToArray();// finalize the lazy linq so it is executed only once
```

Comment:
Did you find that this changed helped? My profiling tests don't show this as being a source of eating up cycles.

Reply:
I found the opposite.  Linq is lazy.  Since the statement is not "terminated", it will execute the linq statement every iteration (which in this case is extremely hot)